Debian avoid unconditional ibacm start

cpaelzer opened the pull request:
This is a successor to the abandoned #393

Per Discussion on the [Mailing list](https://www.spinics.net/lists/linux-rdma/msg70203.html) and some iterations on the [Launchpad bug](https://bugs.launchpad.net/ubuntu/+source/rdma-core/+bug/1794825) I think we should drop the sysV script.

For now it seems the most clear way to get rid of unconditionally starting ibacm (including failing to do so depending on the environment and breaking the postinst actions) due to debhelper actions and IMHO I don't think pre-systemd backports are important for the .deb packages anymore.

I'm open to suggestions how to convince dh_* otherwise to do the right thing, but please check out the [LP bug](https://bugs.launchpad.net/ubuntu/+source/rdma-core/+bug/1794825) about the most obvious approaches that were already tried.

cpaelzer commented:
Branch [debian-avoid-unconditional-ibacm-start-noremove](https://github.com/cpaelzer/rdma-core/tree/debian-avoid-unconditional-ibacm-start-noremove) is another failed approach to show that without removing sysV it doesn't work. Results are discussed on  [LP](https://bugs.launchpad.net/ubuntu/+source/rdma-core/+bug/1794825/comments/16).

The last option would be to manually code the snippets correctly to have:
- sysV init installed but not started
- .service installed but not started
- .socket installed and started

But coding those manually is generally bad and I'd clearly prefer dropping the sysV.
So unless asked by you to do so I'd keep up the PR as it is now.

jgunthorpe commented:
@bdrung up to you on dropping sysvinit...

jgunthorpe commented on debian/changelog:
```diff
@@ -1,3 +1,15 @@
+    - d/ibacm.maintscript: remove old conffile
+    - d/ibacm.install: no more install sysV script
+    - d/not-installed: mark etc/init.d/ibacm as deliberately not installed
+
```
We are not maintaining the changelog here, it is kept in the debian branch for debain. This just has one entry with the right version to make it build right. So drop this change

jgunthorpe commented:
OH and rebase this to lastest master so CI passes, thanks

cpaelzer commented:
Updated per former requests - changes:
- move LP bug reference out of subject
- prefix change with "debian:"
- drop changelog (the commit message body follows a format that can be reused for the changelog if needed)
- rebased on latest master

cpaelzer commented:
Hmm, the current travis build issue
`dh_install: etc/init.d/ibacm exists in debian/tmp but is not installed to anywhere`
should be (and in my own builds is) addressed by [this section](https://github.com/linux-rdma/rdma-core/pull/401/commits/db79920bf649de44f2a3759a08136d8ca3ee0b9e#diff-a75fc9755cbc8a00151ede51c03aee2e) of the commit.

cpaelzer commented:
FYI: I also reached out to [Ubuntu-devel](https://lists.ubuntu.com/archives/ubuntu-devel/2018-October/040525.html) in case there is a "prior art" we could base this on other than removeing the sysV file. But so far no one was able to tell me a good way to do it.

@bdrung - I see @jgunthorpe asked you to make the call on this - any opinion/discussion on this?

cpaelzer commented:
FYI: I also reached out to [Ubuntu-devel](https://lists.ubuntu.com/archives/ubuntu-devel/2018-October/040525.html) in case there is a "prior art" we could base this on other than removeing the sysV file. But so far no one was able to tell me a good way to do it.

@bdrung - I see @jgunthorpe asked you to make the call on this - any opinion/discussion on this?

cpaelzer commented:
FYI: I also reached out to [Ubuntu-devel](https://lists.ubuntu.com/archives/ubuntu-devel/2018-October/040525.html) in case there is a "prior art" we could base this on other than removeing the sysV file. But so far no one was able to tell me a good way to do it.

@bdrung - I see @jgunthorpe asked you to make the call on this - any opinion/discussion on this?

cpaelzer commented:
FYI: I also reached out to [Ubuntu-devel](https://lists.ubuntu.com/archives/ubuntu-devel/2018-October/040525.html) in case there is a "prior art" we could base this on other than removing the sysV file. But so far no one was able to tell me a good way to do it.

@bdrung - I see @jgunthorpe asked you to make the call on this - any opinion/discussion on this?

cpaelzer commented:
FYI: I also reached out to [Ubuntu-devel](https://lists.ubuntu.com/archives/ubuntu-devel/2018-October/040525.html) in case there is a "prior art" we could base this on other than removing the sysV file. But so far no one was able to tell me a good way to do it.

@bdrung - I see @jgunthorpe asked you to make the call on this - any opinion/discussion on this?

cpaelzer commented:
FYI: I also reached out to [Ubuntu-devel](https://lists.ubuntu.com/archives/ubuntu-devel/2018-October/040525.html) in case there is a "prior art" we could base this on other than removing the sysV file. But so far no one was able to tell me a good way to do it.

@bdrung - I see @jgunthorpe asked you to make the call on this - any opinion/discussion on this?

cpaelzer commented:
FYI: I also reached out to [Ubuntu-devel](https://lists.ubuntu.com/archives/ubuntu-devel/2018-October/040525.html) in case there is a "prior art" we could base this on other than removing the sysV file. But so far no one was able to tell me a good way to do it.

@bdrung - I see @jgunthorpe asked you to make the call on this - any opinion/discussion on this?

cpaelzer commented:
FYI: I also reached out to [Ubuntu-devel](https://lists.ubuntu.com/archives/ubuntu-devel/2018-October/040525.html) in case there is a "prior art" we could base this on other than removing the sysV file. But so far no one was able to tell me a good way to do it.

@bdrung - I see @jgunthorpe asked you to make the call on this - any opinion/discussion on this?

cpaelzer commented:
FYI: I also reached out to [Ubuntu-devel](https://lists.ubuntu.com/archives/ubuntu-devel/2018-October/040525.html) in case there is a "prior art" we could base this on other than removing the sysV file. But so far no one was able to tell me a good way to do it.

@bdrung - I see @jgunthorpe asked you to make the call on this - any opinion/discussion on this?

cpaelzer commented:
FYI: I also reached out to [Ubuntu-devel](https://lists.ubuntu.com/archives/ubuntu-devel/2018-October/040525.html) in case there is a "prior art" we could base this on other than removing the sysV file. But so far no one was able to tell me a good way to do it.

@bdrung - I see @jgunthorpe asked you to make the call on this - any opinion/discussion on this?

cpaelzer commented:
Nice, the [GH outage]( https://blog.github.com/2018-10-21-october21-incident-report) made all failed post attempts show up :-/
Sorry for the noise

jgunthorpe commented:
Looks like the old trusty debhelper does not support the not-installed file. I changed it to use INST_EXCLUDE and updated the PR

@bdrung  Yes or No on this?
